presto-main/src/main/java/com/facebook/presto/execution/scheduler/FixedBucketNodeMap.java

```diff
@@ -41,7 +41,10 @@ public FixedBucketNodeMap(ToIntFunction<Split> splitToBucket, List<InternalNode>
     @Override
     public Optional<InternalNode> getAssignedNode(int bucketedId)
     {
-        return Optional.of(bucketToNode.get(bucketedId));
+        if (bucketToNode.size() > bucketedId) {
```

Comment:
It is not clear to me that this is the right fix. If it is, a simpler way is to use `Optional.ofNullable`:

```
    @Override
    public Optional<InternalNode> getAssignedNode(int bucketedId)
    {
        return Optional.ofNullable(bucketToNode.get(bucketedId));
    }
```